Rubberduck.CodeAnalysis/Inspections/Concrete/ImplicitlyTypedConstInspection.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Inspections.Results;
+using Rubberduck.Parsing.Symbols;
+
+namespace Rubberduck.CodeAnalysis.Inspections.Concrete
+{
+    public sealed class ImplicitlyTypedConstInspection : InspectionBase
+    {
+        public ImplicitlyTypedConstInspection(RubberduckParserState state)
+            : base(state) { }
+
+        protected override IEnumerable<IInspectionResult> DoGetInspectionResults()
+        {
+            var declarationFinder = State.DeclarationFinder;
+
+            var implicitlyTypedConsts = declarationFinder.AllDeclarations
+                .Where(declaration => declaration.DeclarationType == DeclarationType.Constant
+                    && declaration.AsTypeContext == null);
+
+            return implicitlyTypedConsts.Select(Result);
+        }
+
+        private IInspectionResult Result(Declaration declaration)
+        {
+            return new IdentifierReferenceInspectionResult(
+                this,
+                declaration.DescriptionString,
+                State,
+                new IdentifierReference(
```

Comment:
Outside of the resolver, references should never be newed up. (I think we should make the constructor internal.)
Since you deal with declarations, use a `DeclarationInspectionResult`.